torch/compiler/__init__.py

```diff
@@ -216,18 +217,69 @@ def assume_constant_result(fn):
 
 def disable(fn=None, recursive=True):
     """
-    This function provides both a decorator and a context manager to disable compilation on a function
+    This function provides a decorator manager to disable compilation on a function
     It also provides the option of recursively disabling called functions
 
+    .. note::
+        Interaction between `compile`, `disable`, and `enable`
+
+        `compile` is is a "marker" that Dynamo should attempt to compile the function
+        and its nested calls.
+
+        `disable` is higher-priority - it signifies that a function (and its nested
+        calls in the case recursive=True) should not be compiled.
+
+        In particular, `disable` overrides `compile` - if you want to re-enable compilation,
+        use `enable`. `disable` and `enable` have the same priority.
+
+        e.g.
+
+            @enable
+            def a(x):
+                ...
+
+            @disable
+            def b(x):
+                a(x)
+                ...
+
+            @compile
+            def c(x):
+                b(x)
+                ...
+
+            @disable
+            def d(x):
+                c(x)
+                ...
+
+        Calling `a`  will result in no compilation.
+        Calling `b` will result in no compilation
+        Calling `c` will result in `c` and `a` being compiled.
+            A graph break will occur when `b` is called.
+        Calling `d` will result in `a` being compiled.
+
     Args:
         fn (optional): The function to disable
         recursive (optional): A boolean value indicating whether the disabling should be recursive.
+
     """
     import torch._dynamo
 
     return torch._dynamo.disable(fn, recursive)
 
 
+def enable(fn=None):
+    """
+    Decorator to re-enable compilation - inverse of `disable`.
+
+    Compilation will only occur if there was a previous `compile` call.
```

Comment:
I think this docstring can be better. At minimum, it can explicitly discuss some of the other use cases we have for enable; e.g., if you have a kernel that you want always compiled because you're using PT2 for kernel authoring

Reply:
Added always-compile kernel example to docs